File: C++/Sorting techniques/Selection sort.c
#include<stdio.h>
#include<conio.h>

void Display(int A[],int n)
{
	int i;
	for(i=0;i<n-1;i++)
	{
		printf("%d ",A[i]);
	}
	printf("\n");
}

void Swap(int *x,int *y)
{
	int temp = *x;
	*x = *y;
	*y = temp;
}

void Selection_sort(int A[],int n)
{
	int i,j,k;
	for(i=0;i<n-1;i++)  // for passes 
	{
		for(j=k=i;j<n;j++)
		{
			if (A[j]<A[k])
			{   k=j; }
		}
		Swap(&A[i],&A[k]);  // After j goes to the end, k holds the minimum element, shift k and i
	}
}

int main()
{
	int A[] = {10,30,20,5,11,100,90,45};
	int n = 8;

	printf("The elements before sorting : ");
	Display(A,n);

	Selection_sort(A,n);

	printf("The elements after sorting : ");
	Display(A,n);

	return 0;
}

File: Python/Sorting techniques/Selection_sort.py
def Selection_sort(list1):
    j=k=0
    for i in range(len(list1)-1):
        j=k=i;
        for j in range(j,len(list1)):
            if list1[j]<list1[k]:
                k=j
        list1[i],list1[k] = list1[k],list1[i]
        
list1 = [2,8,7,4,5]
print("list before sorting : ",*list1)
Selection_sort(list1)
print("list after sorting : ",*list1)
